float32 theta1
float32 theta2
float32 theta3
float32 alpha
float32 beta
